model-optimizer/extensions/middle/sparse_reshape.py

```diff
@@ -46,26 +46,36 @@ def replace_pattern(self, graph: Graph, match: dict):
         input_shape_value = sparse_reshape.in_port(1).data.get_value()
         output_shape_value = sparse_reshape.out_port(1).data.get_value()
 
+        # establish output shape if value of new shape is given as input
+        new_shape_value = sparse_reshape.in_port(2).data.get_value()
+        if output_shape_value is None and new_shape_value is not None:
+            output_shape_value = new_shape_value
+            if np.count_nonzero(output_shape_value == -1) == 1:
+                elem = np.prod(input_shape_value) // np.prod(new_shape_value[new_shape_value != -1])
+                output_shape_value[output_shape_value == -1] = elem
+
         if input_shape_value is None or output_shape_value is None:
             raise Error("Input shape and output shape values must be defined for node {}".format(sparse_reshape.id))
         if not np.array_equal(input_shape_value, output_shape_value):
             raise Error("Input shape and output shape values must be equal for node {}".format(sparse_reshape.id))
 
         input_data_node1 = sparse_reshape.in_node(0)
         input_data_node2 = sparse_reshape.in_node(1)
-        output_data_node1 = sparse_reshape.out_node(0)
-        output_data_node2 = sparse_reshape.out_node(1)
         graph.remove_edge(input_data_node1.id, sparse_reshape.id)
-        graph.remove_edge(sparse_reshape.id, output_data_node1.id)
         graph.remove_edge(input_data_node2.id, sparse_reshape.id)
-        graph.remove_edge(sparse_reshape.id, output_data_node2.id)
-        merge_data_nodes(graph, output_data_node1, input_data_node1)
-        merge_data_nodes(graph, output_data_node2, input_data_node2)
+        if 0 in sparse_reshape.out_nodes():
+            output_data_node1 = sparse_reshape.out_node(0)
```

Comment:
Why are we using deprecated api and need to use data merger?

Reply:
Fixed